tests/unit/ssg-module/test_build_cpe.py

```diff
@@ -90,3 +92,49 @@ def test_extract_referred_nodes():
 
     assert len(results) == 1
     assert results[0].text == 'Source Code'
+
+
+def test_product_cpes():
```

Comment:
What are tests in this file testing?
Is it the ability to finding a file and translate its contents?
If that's the case, I would suggest to reduce those checks, or to at least compare cpe2 attribute to cpe1 attributes and not to a literal value.
That way, tests will be more indicative regarding what they are testing, and they will be somewhat less dependent on the test data.

Reply:
The tests in this file (`ssg-module/test_build_cpe.py`) are unit tests for classes and functions in `/ssg/build_cpe.py`. Specifically, the 2 test cases added in this PR (`test_product_cpes` and `test_content_cpes`) are testing the `ProductCPEs` class. These test cases are focused on testing the `get_cpe` method. This method has been chosen to be tested because other methods of this class are either private or used only inside the constructor of this class, except the `get_cpe_name` method, but that only uses the `get_cpe` method internally. Basically, the following features of the `get_cpe` method are tested in the tests:
* it accepts both CPE name and CPE ID
* it returns the same `CPEItem` object if a CPE name is passed to the CPE as if a CPE ID is passed to the method
* it returns `CPEItem` instances for CPEs that were defined as so-called "product CPEs" (CPEs defined in current product's `product.yml`)
* it returns `CPEItem` instances for CPEs that were defined as so-called "content CPEs" (CPEs defined under the `applicability` directory)
* the returned `CPEItem` instances contain all the data

So unfortunately, it tests also the constructor, which in turn uses the other methods in the class, which means that almost whole class is tested by this. I think that is probably the cause of your concern. Would you prefer to add other tests cases that would test also other methods?

> or to at least compare cpe2 attribute to cpe1 attributes and not to a literal value.

What do you mean by that?

Reply:
I see, I would prefer if it was more obvious what is going on, the test doesn't indicate at all that it depends on the contents of the "applicability" directory. Ideally, one should be able to make the code to load CPEs from mock product data and from mock applicability data. But this is an issue with the tested code, not with tests.
I would then just ask to put this as a comment to the content test.